File: src/main/java/requestreplyapi/entries/ExtendedEntry.java
package requestreplyapi.entries;

public class ExtendedEntry extends Entry{
    private String entryType;

    public ExtendedEntry(String theDest, int thePort, String entryType) {
        super(theDest, thePort);
        this.entryType = entryType;
    }

    public ExtendedEntry() {
    }

    public String type(){
        return entryType;
    }

    @Override
    public String toString() {
        return "ExtendedEntry{" +
                "entryType='" + entryType + '\'' +
                super.toString() +
                '}';
    }
}

File: ObjectRequestBrokerCS/ObjectRequestBrokerCS/requestreplyapi/entries/ExtendedEntry.cs
namespace requestreplyapi.entries
{
	public class ExtendedEntry : Entry
	{
		private string entryType;

		public ExtendedEntry(string theDest, int thePort, string entryType) : base(theDest, thePort)
		{
			this.entryType = entryType;
		}

		public ExtendedEntry()
		{
		}

		public virtual string type()
		{
			return entryType;
		}

		public override string ToString()
		{
			return "ExtendedEntry{" +
					"entryType='" + entryType + '\'' +
					base.ToString() + '}';
		}
	}

}